worldengine/astar.py

```diff
@@ -158,11 +158,10 @@ def __init__(self, map_data, width, height):
     def get_node(self, location):
         x = location.x
         y = location.y
-        if x < 0 or x >= self.w or y < 0 or y >= self.h:
-            return None
-        d = self.m[(y * self.w) + x]
-
-        return Node(location, d, ((y * self.w) + x))
+        if (0 <= x < self.w) and (0 <= y < self.h):
```

Comment:
I used these
```
timeit.timeit('0 <= 50 < 500 and 0 <= 301 < 300', number=10000000)
timeit.timeit('50 < 0 or 50 > 500 or 301 < 0 or 301 > 300', number=10000000)
```

to profile this. The latter method seems to be faster: 0.907968218998576 vs 0.8031158160010818
So I reverted this piece of code. Pushing after your other comments have been worked on.